lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

```diff
@@ -1388,8 +1388,22 @@ private FoundHits queryHits(
                                                                       sr.ascending ? "asc" : "desc"));
             sort.add(sortOrderWrapper);
         }
-        String indexName = objectType == null ? getAnyIndexPattern() : checkIndex(objectType);
-        String urlPath = "/" + indexName + "/" + getDataTableName() + "/_search";
+
+        StringBuffer indexName = new StringBuffer();
+        for(int ii=0; ii<objectTypes.size(); ii++) {
+
+            String objectType = objectTypes.get(ii);
+
+            indexName.append(
+                    objectType == null ? getAnyIndexPattern() : checkIndex(objectType)
+            );
+
+            if( ii < objectTypes.size()-1 )
+                indexName.append(",");
+        }
+
+
+        String urlPath = "/" + indexName.toString() + "/" + getDataTableName() + "/_search";
```

Comment:
how many types in one call can we support given URL length limitations? Should we throw an error if objectTypes.size() > MAX_TYPES_SUPPORTED

Reply:
Alternatively is there a way to specify the indexes to be searched without putting it in the url?

Reply:
I don't think ES has a way of specifying the multiple types in the body of the message. It also looks like ES does not limit the length of the URL although it is configurable and the default is 4k bytes. 

lib/src/kbasesearchengine/search/ElasticIndexingStorage.java

I have added a check for this default size.

Reply:
Lots of discussion around this here. Some info I've found so far:

https://github.com/elastic/elasticsearch/issues/26360
https://www.elastic.co/guide/en/elasticsearch/reference/5.5/search-multi-search.html

Reply:
Index blacklists are possible: https://www.elastic.co/guide/en/elasticsearch/reference/5.5/multi-index.html

Reply:
To record what we said in the standup:
1) Add a maximum search type size in SearchObjectType (say 50 bytes of utf-8)
2) Add a maximum count of search types to search (say 50) on and throw and exception if it's larger than that number

search type size * search type count << 4096 so we have slop for doing blacklists, preventing sub type searches, etc

Blacklists and preventing sub type searches we'll do in another PR.

Reply:
Completed.